### PR TITLE
Deprecated getFieldIndex with first argument as cloud

### PR DESCRIPTION
**Pull request by kunaltyagi**
Pulled out the changes from #3354

### CODE REVIEW

File: doc/tutorials/content/sources/vfh_recognition/nearest_neighbors.cpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ loadHist (const boost::filesystem::path &path, vfh_model &vfh)
   vfh.second.resize (308);
 
   std::vector <pcl::PCLPointField> fields;
-  getFieldIndex (point, "vfh", fields);
+  getFieldIndex<pcl::VFHSignature308> ("vfh", fields);
```
Comment (taketwo) [inline doc/tutorials/content/sources/vfh_recognition/nearest_neighbors.cpp]:
```suggestion
  pcl::getFieldIndex<pcl::VFHSignature308> ("vfh", fields);
```

**Comment by taketwo** [comment]
We forgot to replace calls to deprecated `getFields(cloud, fields)`. I get looots of warnings.

**Comment by kunaltyagi** [comment]
We should try to make `-Werror` a real possibility sometime in future. That'll get rid of forgetting to replace deprecated usage.

**Comment by taketwo** [comment]
> We should try to make -Werror a real possibility sometime in future.

Absolutely!

**Pull request merged**